docs/design/daemon.md

```diff
@@ -0,0 +1,128 @@
+
+For other uses, see the related [feature request](https://github.com/GoogleCloudPlatform/kubernetes/issues/1518)
+
+## Functionality
+
+The Daemon Controller supports standard API features:
+- create
+  - The spec for daemon controllers has a pod template field.
+  - Using the pod’s node selector field, Daemon controllers can be restricted to operate over nodes that have a certain label. For example, suppose that in a cluster some nodes are labeled ‘database’. You can use a daemon controller to launch a datastore pod on exactly those nodes labeled ‘database’.
+  - Using the pod's node name field, Daemon controllers can be restricted to operate on a specified node.
+  - The spec for pod templates that run with the Daemon Controller is the same as the spec for pod templates that run with the Replication Controller, except there will not be a ‘replicas’ field (exactly 1 daemon pod will be launched per node).
+  - We will not guarantee that daemon pods show up on nodes before regular pods - run ordering is out of scope for this controller. 
+  - The initial implementation of Daemon Controller does not guarantee that Daemon pods show up on nodes (for example because of resource limitations of the node), but makes a best effort to launch Daemon pods (like Replication Controllers do with pods). Subsequent revisions might ensure that Daemon pods show up on nodes, pushing out other pods if necessary.
+  - A daemon controller named “foo” adds a “controller: foo” annotation to all the pods that it creates
+  - YAML example:
+```YAML
+  apiVersion: v1
+  kind: Daemon
+  metadata:
+    labels:
+      name: datastore
+    name: datastore
+  spec:
+    template:
+      metadata:
+        labels:
+          name: datastore-shard
+      spec:
+        node-selector: 
+          name: datastore-node
+        containers:
+          name: datastore-shard
+          image: kubernetes/sharded
+          ports:
+            - containerPort: 9042
+              name: main
+```
+  - commands that get info
+    - get (e.g. kubectl get dc)
+    - describe
+  - Modifiers
+    - delete
+    - stop: first we turn down all the pods controller by the daemon (by setting the nodeName to a non-existed name). Then we turn down the daemon controller.
+    - label
+    - update
+    - Daemon controllers have labels, so you could, for example, list all daemon controllers with a certain label (the same way you would for a Replication Controller).
+  - In general, for all the supported features like get, describe, update, etc, the Daemon Controller works in a similar way to the Replication Controller. However, note that the Daemon Controller and the Replication Controller are different constructs.
+
```

Comment:
I agree you should provide some description of a plausible update process. Even if it managed by deployment controller, I assume we need some way deployment controller tweaks daemon controller to accomplish the update? (Or would deployment controller manage the update directly?)

Also, since one of the main purposes of daemon controller is to guarantee one replica per node, you probably want to kill the old pod before starting the new pod (i.e. the application make break if there are two instances running on the same node).